doc/source/ray-air/tuner.rst

```diff
@@ -0,0 +1,216 @@
+    :end-before: __run_config_end__
+
+See the :class:`RunConfig API reference <ray.air.config.RunConfig>` for more details.
+
+
+How do I specify parallelism?
+-----------------------------
+
+You can specify parallelism via the :class:`TuneConfig <ray.tune.tune_config.TuneConfig>` by setting the following flags:
+
+- `num_samples` which specifies the number of trials to run in total
+- `max_concurrent_trials` which specifies the max number of trials to run concurrently
+
+Note that actual parallelism can be less than `max_concurrent_trials` and will be determined by how many trials
+can fit in the cluster at once (i.e., if you have a trial that requires 16 GPUs, your cluster has 32 GPUs,
+and `max_concurrent_trials=10`, the `Tuner` can only run 2 trials concurrently).
+
+.. literalinclude:: doc_code/tuner.py
+    :language: python
+    :start-after: __tune_parallelism_start__
+    :end-before: __tune_parallelism_end__
+
+Read more about this in :ref:`tune-parallelism` section.
+
+
+How do I specify an optimization algorithm?
+-------------------------------------------
+
+You can specify your hyperparameter optimization method via the :class:`TuneConfig <ray.tune.tune_config.TuneConfig>` by setting the following flags:
+
+- `search_alg` which provides an optimizer for selecting the optimal hyperparameters
+- `scheduler` which provides a scheduling/resource allocation algorithm for accelerating the search process
+
+.. literalinclude:: doc_code/tuner.py
+    :language: python
+    :start-after: __tune_optimization_start__
+    :end-before: __tune_optimization_end__
+
+Read more about this in the :ref:`Search Algorithm <search-alg-ref>` and :ref:`Scheduler <schedulers-ref>` section.
+
+How to analyze results?
+-----------------------
+
+``Tuner.fit()`` generates a ResultGrid object. This object contains metrics, results, and checkpoints
+of each trial. Below is a simple usage example:
+
+.. literalinclude:: doc_code/tuner.py
+    :language: python
+    :start-after: __result_grid_inspection_start__
+    :end-before: __result_grid_inspection_end__
+
+Advanced Tuning
+---------------
+
+Tuners also offer the possibility to apply different data preprocessing steps, as shown in the following snippet.
+
+.. literalinclude:: doc_code/tuner.py
+    :language: python
+    :start-after: __tune_preprocess_start__
+    :end-before: __tune_preprocess_end__
+
+You can also sample different train/validation datasets:
+
+.. literalinclude:: doc_code/tuner.py
+    :language: python
+    :start-after: __tune_dataset_start__
+    :end-before: __tune_dataset_end__
+
+
+Restoring and resuming
+----------------------
+A Tuner regularly saves its state, so that a tuning run can be resumed after being interrupted.
+
+Additionally, if trials fail during a tuning run, they can be retried - either from scratch or
+from the latest available checkpoint.
+
+To restore the Tuner state, you just pass the path to the experiment directory to ``Tuner.restore()``.
```

Comment:
To restore the Tuner state, pass the path to the experiment directory as an argument to ``Tuner.restore(...)``